src/server/graphql/resolvers/root.ts

```diff
@@ -1,14 +1,106 @@
-import { pubsub } from '../graphql'
-import { GraphqlRequestContext, GraphqlPubSubKey } from '../graphql.models'
+import {
+	createGame,
+	getGame,
+	joinGame,
+	leaveGame,
+	playMove,
+	resign,
+	toggleDrawDesire,
+	toggleReady,
+} from '../../gameEngine/gameEngine'
+import { EngineChessGame } from '../../gameEngine/store/store.models'
+import { publishGameStateChange, pubsub } from '../graphql'
+import { GraphqlPubSubKey, GraphqlRequestContext } from '../graphql.models'
 
 const root = {
 	Query: {
-		sessionId: (_root: any, _args: any, ctx: GraphqlRequestContext) =>
-			ctx.sessionId,
+		gameById: (
+			_root: any,
+			args: { gameId: string },
+			_ctx: GraphqlRequestContext,
+		) => {
+			const game = getGame(args.gameId)
+			return game
+		},
+	},
+	Mutation: {
+		createGame(
+			_root: any,
+			_args: any,
+			ctx: GraphqlRequestContext,
+		): EngineChessGame {
+			const game = createGame(ctx.sessionId)
+			return game
+		},
+		joinGame(
+			_root: any,
+			args: { gameId: string },
+			ctx: GraphqlRequestContext,
+		): EngineChessGame {
+			const game = joinGame(ctx.sessionId, args.gameId)
```

Comment:
What about

```suggestion
			const game = getGameFromStore(gameId)
                        joinGame(ctx.sessionId, game)
```

or even
```suggestion
			const game = getGameFromStore(gameId)
                        game.join(ctx.sessionId)
```

?